src/components/TimePicker.tsx

```diff
@@ -0,0 +1,81 @@
+import './styles/time_picker.less'
+
+import React from 'react'
+import ClockIcon from './icons/ClockIcon.svg'
+
+import CustomParseFormat from 'dayjs/plugin/customParseFormat'
+import dayjs, { Dayjs } from 'dayjs'
+import Select from './select'
+import classNames from 'classnames'
+
+dayjs.extend(CustomParseFormat)
+
+const MINUTES_INTERVAL = 5
+
+type Props = {
+    minutesInterval?: number
+    onTimeChanged?: (value: string) => void
+    min?: string
+    max?: string
+    defaultValue?: string
+    className?: string
+    optionsClassName?: string
+}
+
+export default function TimePicker({
+    minutesInterval,
+    onTimeChanged,
+    min,
+    max,
+    defaultValue,
```

Comment:
Why this name? Is this a [controlled](https://reactjs.org/docs/forms.html#controlled-components) or an [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html) component? If it's a controlled component (as it seems to me it is at first sight) then this should be called `value` to follow the convention.

Reply:
I think this comes from the default input element doesn't it? 
![image](https://user-images.githubusercontent.com/25244878/104444078-30a90680-558f-11eb-917b-9ee569487e25.png)

Reply:
Indeed, but the default input element has both `defaultValue` and also a `value` prop. You can use it in one of those two modes: controlled or uncontrolled. When you use it as a controlled input, you use `value`, and `defaultValue` is for initialization purposes only. That is, `defaultValue` only provides the initial value with which it is rendered, and subsequent changes to its value is not managed by React (that is, there's no need for an `onChange` handler).

However, in this case it seems to me that this component only supports the controlled mode. That is, you provide its value as a prop at all times, and the context where this component is rendered is supposed to manage its value as state, and listen to attempts to change it via the `onChange` prop. If this is so, the customary thing to do is to have a `value`/`onChange` pair of props. No need to have a `defaultValue` when it does not support the uncontrolled mode.